spark/src/test/resources/tpcds-plan-stability/approved-plans-v1_4-spark3_5/q1/explain.txt

```diff
@@ -172,12 +172,12 @@ Arguments: [ctr_customer_sk#9, ctr_store_sk#10], [ctr_customer_sk#9, ctr_store_s
 Output [2]: [s_store_sk#24, s_state#25]
 Batched: true
 Location [not included in comparison]/{warehouse_dir}/store]
-PushedFilters: [IsNotNull(s_state), EqualTo(s_state,TN), IsNotNull(s_store_sk)]
+PushedFilters: [IsNotNull(s_state), IsNotNull(s_store_sk)]
 ReadSchema: struct<s_store_sk:int,s_state:string>
```

Comment:
This seems like a regression in Spark. Now that Spark uses `rpad` for this filter, it can no longer be pushed down to the scan.

Related issue: https://issues.apache.org/jira/browse/SPARK-50692